shimmingtoolbox/read_nii.py

```diff
@@ -0,0 +1,84 @@
+import json
+import math
+import nibabel as nib
+import os
+
+
+def read_nii(niiPath):
+    """
+
+    Args:
+        niiPath (str): Path to the nifti file
+
+    Returns:
+        image:
+        info:
+        jsonData:
+    """
+
+    info = nib.load(niiPath)
+    image = info.get_fdata()
+
+    # `extractBefore` should get the correct filename in both.nii and.nii.gz cases
+    jsonPath = niiPath.split('.nii')[0] + '.json'
+
+    if os.path.isfile(jsonPath):
+        jsonData = json.load(open(jsonPath))
+    else:
+        jsonData = []
+        print('Warning: No json file has been found')
+
+    # NOTE: nib.load automatically scales the nifti and replaces scl_inter and scl_slope with 'nan'
+    # See the "Data scaling" section in https://nipy.org/nibabel/nifti_images.html
+    if ('Manufacturer' in jsonData) and (jsonData['Manufacturer'] == 'Siemens') \
+            and (check_json_image_type(jsonData) == 'phase'):
+
+        image, info = convert_siemens_phase(image, info)
+
+    return image, info, jsonData
+
+# -----------------------------------------------------------------------------
+# Local functions
+# -----------------------------------------------------------------------------
+
+
+def check_json_image_type(jsonData):
+    imgType = []
+
+    if not jsonData:
+        print('Warning: The image type has not been found because the json file is empty')
+        return imgType
+
+    assert isinstance(jsonData, dict)  # Check that jsonData is a dictionary
+
+    if 'ImageType' in jsonData:
+        isPhase = "P" in jsonData['ImageType']
+        isMag = "M" in jsonData['ImageType']
+
+        if isPhase and isMag:
+            # Both true: json file and / or DICOM issue(hopefully this doesn't occur?)
+            print('Warning: Ambiguous ImageType entry in json file: Indicates magnitude AND phase?')
```

Comment:
To print I suggest you use the logging module of Python. It is a standard module to print information, warnings and errors to console and wrap everything that you need to log (.log files and console)

Reply:
also, a fairly common approach is to use `print()` in CLIs and `logging` in APIs (which is the latter here)

Reply:
There's uh, also,

```
import warnings
```

 ```suggestion
            warnings.warn("Ambiguous ImageType entry in json file: Indicates magnitude AND phase?')
```

But I think I would promote this case to an exception

```suggestion
            raise ValueError("Ambiguous ImageType entry in json file: Indicates magnitude AND phase?')
```

Python is all about crashing early and behaving predictably instead of trying to guess its way through.

Reply:
[This essay](https://dave.cheney.net/2015/11/05/lets-talk-about-logging) gave me good reflections on this. It advocates for only having three levels: info, debug, and fatal. In python, that means `print()` or `logging.info()` (depending on whether you're a CLI or a library like Julien says), `logging.debug()`, and `raise <some exception>`. I think it's right.

In this case, I think I would do

```suggestion
            raise ValueError('Ambiguous ImageType entry in json file: Indicates magnitude AND phase?')
```

Reply:
It's true for most of the programming languages. You should raise an exception for every case that the code have to guess for himself and try somehting odd instead of a clear path prewritten by the developper.